### PR TITLE
Zen 599 sidebar scroll

### PR DESCRIPTION
**Pull request by subaya710**
**Ticket**: [ZEN-599](https://jira.simpleviewtools.com/browse/ZEN-599)

## Context

When there are many items added to the side-out menu , they pass beyond the bottom edge of the screen. We are not able to view or interact with items at the bottom since there is no scroll bar.

## Goal

Provide a scrollbar to the side-out menu as well as appropriate bottom padding so that all items can be accessed.

## Updates

- repos/tap/components/sidebar/content/fileTreePanel.js
Added a new view/div that wraps around the menu child items . Added a scrollbar to this element along with reStyle with the necessary height and bottom padding

## Testing

Docker image : https://github.com/simpleviewinc/keg-herkin/pkgs/container/keg-herkin

**Setup**

- Pull this PR => keg herkin && keg pr 42
- Navigate to repos/example folder and then start keg herkin => keg herkin start
- Open launched app at http://herkin-zen-599-sidebarscroll.local.keghub.io/
- Escape from the pop up window so you are on the main site

**Test**

- Open the side bar menu and expand all sub sections 
- Add feature 10-15 feature files by clicking on the plus symbol in the 'test files' header
- The items towards the bottom should have disappeared from view
- hover over the sidebar menu and scroll. 
- you should be able to interact with items in the menu, even items that have moved below the bottom page boundary

### CODE REVIEW

**Review by lancetipton — CHANGES_REQUESTED** [review]
Overall looks good, but there are a few issue to resolve.
Along with the other github comments, the UI has a few issues
1. Odd space between the menu and scroll => https://i.imgur.com/9yVfLoG.png
   1. Not a huge deal, but doesn't look very professional
   2. Would be better if it aligned with the edge of the side-menu
2. The Arrow toggle gets cut-off when hovered => https://i.imgur.com/pDQmnGw.png
   1. It looks like the overflow hidden on some parent element is causing this
3. Empty space below the menu => https://i.imgur.com/OKYnNQ6.png
   1. We should make this white, so it looks like the menu takes up the entire height

Let me know if you have any questions. Thanks

**Comment by subaya710** [comment]
Changes requested have been made . Adding a new wrapper div and styling it as required solves the 3 issues reported in the review

File: repos/tap/theme/components/sidebar/sidebar.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ export const sidebar = theme => ({
   main: {
     zIndex: 5,
     left: 250,
-    position: 'fixed',
+    position: 'fixed',
```
Comment (lancetipton) [inline repos/tap/theme/components/sidebar/sidebar.js:7]: Remove the extra space at the end of this line

File: repos/tap/components/sidebar/content/fileTreePanel.js [hunk 2]
```diff
@@ -49,7 +49,7 @@ const FileTreeHeader = ({styles, title}) => {
     </View>
   )
 }
-
+const treeScroll = {overflowY: "scroll", height: "100vh", paddingBottom: "9em" }
```
Comment (lancetipton) [inline repos/tap/components/sidebar/content/fileTreePanel.js:52]: This should not be defined inline like this.
Either move it to a theme file, or use `reStyle` from retheme

File: repos/tap/components/sidebar/content/fileTreePanel.js [hunk 3]
```diff
@@ -59,13 +59,16 @@ const FileTreeHeader = ({styles, title}) => {
 export const FileTreePanel = (props) => {
   const theme = useTheme()
   const styles = theme.get('sidebar.content')
+  console.log(styles)
```
Comment (lancetipton) [inline repos/tap/components/sidebar/content/fileTreePanel.js:62]: You already mentioned you would remove this, so I'm just putting a comment here for so I remember to check it

**Review by lancetipton — CHANGES_REQUESTED** [review]
I didn't run the code, but there are a few updates that are needed. resubmit the PR after making the updates, and I'll run it then.

Thanks

**Review by lancetipton — APPROVED** [review]
Excellet work. Looks good, thanks

**Review by mpcarolin — APPROVED** [review]
Looks good. Thanks!

If you can figure out the dynamic `paddingBottom`, that would be an awesome stretch goal, but no worries if not.

**Pull request merged**